gel-cli-instance/src/local/localbackup.rs

```diff
@@ -30,13 +31,40 @@ use super::LocalInstanceHandle;
 const BACKUP_TIMEOUT: Duration = Duration::from_secs(60 * 60);
 const BACKUP_LIVENESS_INTERVAL: Duration = Duration::from_secs(60);
 
+#[derive(Clone)]
 pub struct LocalBackup {
+    pg_backup: Arc<PgBackupCommands<SystemProcessRunner>>,
     handle: LocalInstanceHandle,
 }
 
 impl LocalBackup {
-    pub fn new(handle: LocalInstanceHandle) -> Self {
-        Self { handle }
+    pub fn new(handle: LocalInstanceHandle) -> Result<Self, InstanceOpError> {
+        let pg_backup = PgBackupCommands::new(SystemProcessRunner, handle.bin_dir.clone());
+
+        if !pg_backup.has_all_executables() {
+            return Err(InstanceOpError::Unsupported(
+                "older releases: missing pg_basebackup, pg_combinebackup, or pg_verifybackup"
+                    .to_string(),
+            ));
+        }
+
+        // Some 6.x releases appear to have a broken pg_hba.conf. Check for that and disable backups.
+        let pg_hba = handle.paths.data_dir.join("pg_hba.conf");
+        let Ok(pg_hba_contents) = std::fs::read_to_string(&pg_hba) else {
+            return Err(InstanceOpError::Unsupported(
+                "older releases: pg_hba.conf not found or unreadable".to_string(),
+            ));
+        };
+        if !pg_hba_contents.contains("local replication postgres trust") {
+            return Err(InstanceOpError::Unsupported(
+                "older releases: pg_hba.conf does not allow local replication connections (add `local replication postgres trust` to the file)".to_string(),
```

Comment:
Should we consider doing this ourselves? Maybe at least printing the path to the file would be helpful.

Reply:
I considered doing out ourselves, but restarting postgres is tricky. I can print out the full path though.

Reply:
New error:

```
Automatic backup is disabled.
Unsupported operation for older releases, pg_hba.conf does not allow local replication connections (add `local replication postgres trust` to "/home/matt/.local/share/edgedb/data/dump_restore2/pg_hba.conf")
```